conans/client/cmd/uploader.py

```diff
@@ -29,8 +29,9 @@ def __init__(self, client_cache, user_io, remote_manager, registry):
 
     def upload(self, reference_or_pattern, package_id=None, all_packages=None,
                force=False, confirm=False, retry=0, retry_wait=0, skip_upload=False,
-               integrity_check=False, no_overwrite=None, remote_name=None):
+               integrity_check=False, no_overwrite=None, remote_name=None, recorder=None):
```

Comment:
It is very weird to accept a recorder=None, and fail if it is not defined.
But leave it as-is, no big issue.